lib/src/linear_feedback_shift_register.dart

```diff
@@ -0,0 +1,114 @@
+// Copyright (C) 2023 Intel Corporation
+
+// SPDX-License-Identifier: BSD-3-Clause
+
+//
+
+// linear_feedback_shift_register.dart
+
+// Implementation of Galois Linear Feedback Shift Register.
+
+//
+
+// 2024 October 1
+
+// Author: Omonefe Itietie <omonefe.itietie@intel.com>
+
+//
+
+import 'dart:collection';
+import 'package:rohd/rohd.dart';
+
+/// Galois Linear Feedback Shift Register
+class LinearFeedbackShiftRegister extends Module {
+  /// Contains polynomial size for LFSR
+  final int width;
+
+  /// Contains seed of LFSR (starting value)
+  Logic state;
+
+  /// Data names for signals
+  final String dataName;
+
+  /// Contains bit string that will be used to calculate the output
+  final Logic taps;
+
+  /// Output for shift register
+  Logic get dataOut => output('${dataName}_out');
+
+  /// The number of stages in this shift register.
+  final int shifts;
+
+  /// A [List] of [output]s where the `n`'th entry corresponds to a version of
+  /// the input data after passing through `n + 1` flops.
+  late final List<Logic> stages = UnmodifiableListView(
+      [for (var i = 0; i < shifts; i++) output(_stageName(i))]);
+
+  /// The name of the signal (and output pin) for the [i]th stage.
+  String _stageName(int i) => '${dataName}_stage_$i';
+
+  LinearFeedbackShiftRegister(Logic dataIn,
+      {required Logic clk,
+      required this.state,
+      required this.shifts,
+      required this.taps,
```

Comment:
Careful using `this.` for inputs of a module, since we want to replace them using `addInput`